Source: oahong@oahong.me
Upstream: no, this is not a proper fix
Reason: Fix build system
diff -urN supertuxkart-0.7.1b~/configure.ac supertuxkart-0.7.1b/configure.ac
--- supertuxkart-0.7.1b~/configure.ac	2011-04-21 07:33:33.000000000 +0800
+++ supertuxkart-0.7.1b/configure.ac	2011-05-29 16:44:04.930801294 +0800
@@ -402,7 +402,6 @@
   enet_LIBS="-Lenet -lenet"
   ;;
 esac
-ENETTREE="src/enet"
 
 # =================
 # Disable/enable addons support
@@ -412,6 +411,7 @@
 if test "x$with_addons" == "xyes" ; then
     SUMMARY="$SUMMARY\nEnabling addons support."
     AC_DEFINE(ADDONS_MANAGER)
+    AM_CONDITIONAL(ADDONS_MANAGER, test x$with_addons = xyes)
 fi
 # ==========================================
 # Check for a known compiler bug, details in 
@@ -433,6 +433,7 @@
 AC_SUBST(fribidi_LIBS)
 AC_SUBST(bullet_LIBS)
 AC_SUBST(enet_LIBS)
+AC_SUBST(enet_CFLAGS)
 AC_SUBST(opengl_LIBS)
 AC_SUBST(openal_LIBS)
 AC_SUBST(oggvorbis_LIBS)
@@ -461,8 +462,7 @@
  src/Makefile                       \
  src/ide/Makefile                   \
  src/bullet/Makefile                \
- src/bullet/src/Makefile            \
- src/enet/Makefile                  \
+ src/bullet/src/Makefile
 ])
 AC_OUTPUT
 echo -e $SUMMARY
diff -urN supertuxkart-0.7.1b~/data/challenges/Makefile.am supertuxkart-0.7.1b/data/challenges/Makefile.am
--- supertuxkart-0.7.1b~/data/challenges/Makefile.am	2011-04-04 08:09:16.000000000 +0800
+++ supertuxkart-0.7.1b/data/challenges/Makefile.am	2011-05-29 17:05:47.142801892 +0800
@@ -1,5 +1,5 @@
 # challenges/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/challenges
+pkgdatadir = $(datadir)/$(PACKAGE)/data/challenges
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.challenge") 
diff -urN supertuxkart-0.7.1b~/data/fonts/Makefile.am supertuxkart-0.7.1b/data/fonts/Makefile.am
--- supertuxkart-0.7.1b~/data/fonts/Makefile.am	2011-04-04 08:09:06.000000000 +0800
+++ supertuxkart-0.7.1b/data/fonts/Makefile.am	2011-05-29 17:05:47.209801892 +0800
@@ -1,6 +1,6 @@
 # data/fonts/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/fonts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/fonts
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.png") \
                      $(shell find $(srcdir) -name "*.xml")
diff -urN supertuxkart-0.7.1b~/data/gfx/Makefile.am supertuxkart-0.7.1b/data/gfx/Makefile.am
--- supertuxkart-0.7.1b~/data/gfx/Makefile.am	2011-04-04 08:09:15.000000000 +0800
+++ supertuxkart-0.7.1b/data/gfx/Makefile.am	2011-05-29 17:05:47.156801892 +0800
@@ -1,5 +1,5 @@
 #data/gfx
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/gfx
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/gfx
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.xml")
 
diff -urN supertuxkart-0.7.1b~/data/grandprix/Makefile.am supertuxkart-0.7.1b/data/grandprix/Makefile.am
--- supertuxkart-0.7.1b~/data/grandprix/Makefile.am	2011-04-04 08:08:59.000000000 +0800
+++ supertuxkart-0.7.1b/data/grandprix/Makefile.am	2011-05-29 17:05:47.215801892 +0800
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/grandprix
+pkgdatadir = $(datadir)/$(PACKAGE)/data/grandprix
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.grandprix")  \
                     $(shell find $(srcdir) -name "*.challenge")  \
diff -urN supertuxkart-0.7.1b~/data/gui/Makefile.am supertuxkart-0.7.1b/data/gui/Makefile.am
--- supertuxkart-0.7.1b~/data/gui/Makefile.am	2011-04-04 08:09:03.000000000 +0800
+++ supertuxkart-0.7.1b/data/gui/Makefile.am	2011-05-29 17:05:47.191801892 +0800
@@ -1,6 +1,6 @@
 # data/gui/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/gui
+pkgdatadir = $(datadir)/$(PACKAGE)/data/gui
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.png"      ) \
                             $(shell find $(srcdir) -name "*.jpg"      ) \
diff -urN supertuxkart-0.7.1b~/data/karts/Makefile.am supertuxkart-0.7.1b/data/karts/Makefile.am
--- supertuxkart-0.7.1b~/data/karts/Makefile.am	2011-04-14 08:04:40.000000000 +0800
+++ supertuxkart-0.7.1b/data/karts/Makefile.am	2011-05-29 17:05:47.184801892 +0800
@@ -1,6 +1,6 @@
 # data/karts
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/karts
+pkgdatadir= $(datadir)/$(PACKAGE)/data/karts
 
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "materials.xml") \
                            $(shell find $(srcdir) -name "*.png")         \
diff -urN supertuxkart-0.7.1b~/data/Makefile.am supertuxkart-0.7.1b/data/Makefile.am
--- supertuxkart-0.7.1b~/data/Makefile.am	2011-04-04 08:09:05.000000000 +0800
+++ supertuxkart-0.7.1b/data/Makefile.am	2011-05-29 17:05:47.221801892 +0800
@@ -1,7 +1,7 @@
 # data/
 
 SUBDIRS    = challenges fonts gfx gui karts models music po sfx textures tracks grandprix
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data
+pkgdatadir = $(datadir)/$(PACKAGE)/data
 
 dist_pkgdata_DATA = \
  $(shell find $(srcdir) -name "*.data")                  \
diff -urN supertuxkart-0.7.1b~/data/models/Makefile.am supertuxkart-0.7.1b/data/models/Makefile.am
--- supertuxkart-0.7.1b~/data/models/Makefile.am	2011-04-04 08:09:05.000000000 +0800
+++ supertuxkart-0.7.1b/data/models/Makefile.am	2011-05-29 17:05:47.203801892 +0800
@@ -1,6 +1,6 @@
 # data/models/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/models
+pkgdatadir = $(datadir)/$(PACKAGE)/data/models
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.b3d") \
                      $(shell find $(srcdir) -name "*.png") \
diff -urN supertuxkart-0.7.1b~/data/music/Makefile.am supertuxkart-0.7.1b/data/music/Makefile.am
--- supertuxkart-0.7.1b~/data/music/Makefile.am	2011-04-04 08:09:07.000000000 +0800
+++ supertuxkart-0.7.1b/data/music/Makefile.am	2011-05-29 17:05:47.177801892 +0800
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/music
+pkgdatadir = $(datadir)/$(PACKAGE)/data/music
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.ogg")  \
                     $(shell find $(srcdir) -name "*.music") \
diff -urN supertuxkart-0.7.1b~/data/po/Makefile.am supertuxkart-0.7.1b/data/po/Makefile.am
--- supertuxkart-0.7.1b~/data/po/Makefile.am	2011-04-04 08:08:36.000000000 +0800
+++ supertuxkart-0.7.1b/data/po/Makefile.am	2011-05-29 17:05:47.170801892 +0800
@@ -1,6 +1,6 @@
 # data/po/
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/data/po
+pkgdatadir = $(datadir)/@PACKAGE@/data/po
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.po") \
                     supertuxkart.pot
diff -urN supertuxkart-0.7.1b~/data/sfx/Makefile.am supertuxkart-0.7.1b/data/sfx/Makefile.am
--- supertuxkart-0.7.1b~/data/sfx/Makefile.am	2011-04-04 08:09:04.000000000 +0800
+++ supertuxkart-0.7.1b/data/sfx/Makefile.am	2011-05-29 17:05:47.149801892 +0800
@@ -1,6 +1,6 @@
 # sfx/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/sfx
+pkgdatadir = $(datadir)/$(PACKAGE)/data/sfx
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ogg")      \
                             $(shell find $(srcdir) -name "sfx.xml")    \
diff -urN supertuxkart-0.7.1b~/data/textures/Makefile.am supertuxkart-0.7.1b/data/textures/Makefile.am
--- supertuxkart-0.7.1b~/data/textures/Makefile.am	2011-04-04 08:09:15.000000000 +0800
+++ supertuxkart-0.7.1b/data/textures/Makefile.am	2011-05-29 17:05:47.164801892 +0800
@@ -1,6 +1,6 @@
 # data/textures
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/textures
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/textures
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.png") \
                     $(shell find $(srcdir) -name "*.jpg") \
                     materials.xml License.txt
diff -urN supertuxkart-0.7.1b~/data/tracks/Makefile.am supertuxkart-0.7.1b/data/tracks/Makefile.am
--- supertuxkart-0.7.1b~/data/tracks/Makefile.am	2011-04-04 08:08:51.000000000 +0800
+++ supertuxkart-0.7.1b/data/tracks/Makefile.am	2011-05-29 17:05:47.197801892 +0800
@@ -1,6 +1,6 @@
 # data/tracks
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/tracks
+pkgdatadir= $(datadir)/$(PACKAGE)/data/tracks
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.jpg")           \
                            $(shell find $(srcdir) -name "*.JPG")           \
                            $(shell find $(srcdir) -name "*.png")           \
diff -urN supertuxkart-0.7.1b~/Makefile.am supertuxkart-0.7.1b/Makefile.am
--- supertuxkart-0.7.1b~/Makefile.am	2011-04-04 08:08:35.000000000 +0800
+++ supertuxkart-0.7.1b/Makefile.am	2011-05-29 16:16:20.651800520 +0800
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = $(BULLETTREE) $(ENETTREE) src doc data
+SUBDIRS = $(BULLETTREE) src doc data
 
 pkgdatadir=$(datadir)/doc/$(PACKAGE)
 dist_pkgdata_DATA=AUTHORS ChangeLog COPYING README
diff -urN supertuxkart-0.7.1b~/src/Makefile.am supertuxkart-0.7.1b/src/Makefile.am
--- supertuxkart-0.7.1b~/src/Makefile.am	2011-04-13 13:57:03.000000000 +0800
+++ supertuxkart-0.7.1b/src/Makefile.am	2011-05-29 17:15:20.494802159 +0800
@@ -2,11 +2,10 @@
 
 SUBDIRS = ide
 
-bindir=$(prefix)/games
 bin_PROGRAMS = supertuxkart
 
-AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
- -I$(srcdir)/bullet/src/ -I$(srcdir)/enet/include/
+AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/$(PACKAGE)/\""  \
+ -I$(srcdir)/bullet/src/ $(enet_CFLAGS)
 
 supertuxkart_SOURCES = \
  main.cpp \
@@ -450,3 +449,6 @@
         $(irrlicht_LIBS) $(fribidi_LIBS) $(bullet_LIBS) $(enet_LIBS) \
         $(opengl_LIBS) $(openal_LIBS) $(oggvorbis_LIBS) \
         $(INTLLIBS) $(LIBCURL_LIBS) $(LIBCURL_CFLAGS)
+if ADDONS_MANAGER
+supertuxkart_LDADD += -lpthread
+endif
